digicampipe/calib/camera/charge.py

```diff
@@ -135,3 +135,21 @@ def compute_photo_electron(events, gains):
         event.data.reconstructed_number_of_pe = pe
 
         yield event
+
+
+def compute_sample_photo_electron(events, gains_integral):
+    """
+    :param events: a stream of events
+    :param gains_integral: value corresponding to how many integrated
+    adc counts (baseline subtracted) over n_sample_integral corresponds to 1 pe
+    :return: a stream of event with event.data.sample_pe filled with
+    fractional pe for each pixel and each sample. Integrating the
+    fractional pe along all samples gives the charge in pe of the
+    full event.
+    """
+    for count, event in enumerate(events):
+        adc_samples = event.data.adc_samples
+        gain_drop = event.data.gain_drop[:, None]
+        sample_pe = adc_samples / (gains_integral[:, None] * gain_drop)
```

Comment:
The idea was to avoid the smoothing of the integration over 7 bins.
I think to be rigorous, one should do a deconvolution with the pulse template, but I did not want to do that either as deconvolutions can fail badly.